firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -80,6 +80,9 @@ void homeAxis(int motor[])
     delay(1);
     digitalWrite(motor[STEP_PIN], HIGH);
   }
+
```

Comment:
So this function is a bit special, since it passes in a specific motor. That being said, we need to differentiate between the motor arrays passed in, which we can do by comparing addresses. That comparison allows us to figure out what motor we're dealing with. We should also have a check for null in case it's passed through by accident, since we're dealing with addresses.

Reply:
However, this only allows us to figure out what motor we're dealing with. We need to add to either of two possible variables, currentX and currentY. To do that, we'll use a pointer! If you need help lmk, cuz pointers are wack. But essentially, we'll make the temp variable hold the address of one of those variables, which then lets us deal with it arbitrarily. I have the code below.

Reply:
```c
void homeAxis(int motor[])
{
  // This is where you need to put these extra variables
  int * currentMotorPos;
  int currentMotorDir, currentMotorNumEighthSteps;
  int i;

  currentMotorPos = (motor == xMotor) ? &currentX : &currentY;

  // Do all of the motor control stuff
  // =====================
  // Here's one while loop for example
  // This one uses whole steps (8 1/8 steps)
  // and moves to the left
  currentMotorNumEighthSteps = 8;
  currentMotorNumEighthSteps *= -1;
  while (condition)
  {
    // Above this will be the code triggering the pulse to the motor driver
    *currentMotorPos += currentMotorNumEighthSteps;
  }

  // Set the motor position to "home" position
  *currentMotorPos = 0;  
}
```

Reply:
Did I implement this correctly?

Reply:
Functionally, it looks perfect! Nashir can double check before we fully merge but yeah it looks awesome. Thanks!